tables/audisp/src/fileeventstableplugin.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <zeek/iaudispconsumer.h>
+#include <zeek/ivirtualtable.h>
+#include <zeek/izeekconfiguration.h>
+#include <zeek/izeeklogger.h>
+
+namespace zeek {
+/// \brief Provides the file_events table
+class FileEventsTablePlugin final : public IVirtualTable {
+  struct PrivateData;
+  std::unique_ptr<PrivateData> d;
+
+public:
+  /// \brief Factory method
+  /// \param obj Where the created object is stored
+  /// \param configuration An initialized configuration object
+  /// \param logger An initialized logger object
+  /// \return A Status object
+  static Status create(Ref &obj, IZeekConfiguration &configuration,
+                       IZeekLogger &logger);
+
+  /// \brief Destructor
+  virtual ~FileEventsTablePlugin() override;
+
+  /// \return The table name
+  virtual const std::string &name() const override;
+
+  /// \return The table schema
+  virtual const Schema &schema() const override;
+
+  /// \brief Generates the row list containing the fields from the given
+  ///        configuration object
+  /// \param row_list Where the generated rows are stored
+  /// \return A Status object
+  virtual Status generateRowList(RowList &row_list) override;
+
+  /// \brief Processes the specified event list, generating new rows
+  /// \param event_list A list of Audit events
+  /// \return A Status object
+  Status processEvents(const IAudispConsumer::AuditEventList &event_list);
+
+protected:
+  /// \brief Constructor
+  /// \param configuration An initialized configuration object
+  /// \param logger An initialized logger object
+  FileEventsTablePlugin(IZeekConfiguration &configuration, IZeekLogger &logger);
+
+  /// \brief Combines working directory with file path
+  /// \param cwd current directory path
+  /// \param path file path
+  static std::string CombinePaths(const std::string &cwd,
```

Comment:
Do we need as a member or even in the public interface? Otherwise we could just create a standalone, `static` function in the impl file wrapped in an anonymous namespace to make it truly private.

Reply:
Most of the classes adopt static methods in the classes so that tests can be implemented in a more clean way

Reply:
As a reason for making static function is mentioned by Alessandro, I am marking this as resolve as well.

Reply:
We are adding a function to a public interface because of test code we have not written. I am not sure how an a `static` method helps with testing when we could just as well provide a decoupled function in some private `detail` namespace. Additionally, we could always expose functionality later when we actually need it (in tests or elsewhere).

I am leaving this up to you @Wajihulhassan 🤷.